### PR TITLE
yumpkg module: Add support for key_url parameter when adding repos

### PR DESCRIPTION
**Pull request by bbczeuz**
### What does this PR do?
The `pkgrepo` state module's `managed` function has a `key_url` parameter. This is currently implemented for deb repos, but silently ignored for rpm repos. The PR implements key downloading and importing for rpm based systems.

### What issues does this PR fix or reference?
#49516

### Previous Behavior
The `key_url` parameter is silently ignored on rpm systems

### New Behavior
The file at `key_url` is imported using 'rpm --import'

### Tests written?
Manually using sls (Centos7):
```
add_influxdata_repo:
  pkgrepo.managed:
    - name: influxdb
    - humanname: influxdata
    - baseurl: https://repos.influxdata.com/rhel/\$releasever/\$basearch/stable
    - key_url: salt://files/influxdb.key
    - gpgcheck: 1
```

### Commits signed with GPG?

No

### CODE REVIEW

**Comment by bbczeuz** [comment]
The implementation is basically an adapted version from the aptpkg module

**Comment by bbczeuz** [comment]
* In theory, we could implement code to avoid importing the same key everytime the state runs by comparing known keys to the new one, but this is the same as the apt-version does.
* The key file is not being stored at the usual location (/etc/pki/rpm-gpg/), but directly imported.

**Review by terminalmage — APPROVED** [review]

File: salt/modules/yumpkg.py [hunk 1]
```diff
@@ -2848,6 +2850,22 @@ def mod_repo(repo, basedir=None, **kwargs):
...
+        fn_ = __salt__['cp.cache_file'](key_url, saltenv=(kwargs['saltenv'] if 'saltenv' in kwargs else 'base'))
+        if not fn_:
+            raise CommandExecutionError(
+                'Error: file not found: {0}'.format(key_url)
+            )
+        cmd = ['rpm', '--import', fn_]
+        out = __salt__['cmd.retcode'](cmd, python_shell=False, **kwargs)
+        if out != 0:
+            raise CommandExecutionError(
+                'Error: failed to add key from {0}'.format(key_url)
```
Comment (isbm) [inline salt/modules/yumpkg.py]: It raises a question: "What key, again?". Let be more specific here.

**Review by isbm — CHANGES_REQUESTED** [review]

**Comment by bbczeuz** [comment]
@isbm : thanks for the review. Messages clarified.

**Review by isbm — CHANGES_REQUESTED** [review]

**Comment by bbczeuz** [comment]
@isbm Done, thanks again :)

**Comment by isbm** [comment]
> @isbm Done, thanks again :)

Next time don't just copypaste my mistakes. :stuck_out_tongue: Thanks!

**Review by isbm — APPROVED** [review]

**Comment by SooLFaa** [comment]
Use CVE-2018-17027 
Use CVE-2018-17028 
for vulnerability auth bypass

**Comment by bbczeuz** [comment]
@SooLFaa Can you elaborate on how this is relevant to the PR?

**Comment by SooLFaa** [comment]
Please send this ticket, the one who is responsible for security.14.09.2018, 13:34, "Claudius Zingerli" <notifications@github.com>:@SooLFaa Can you elaborate on how this is relevant to the PR?

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.

-- С уважением, Алексей!

**Pull request merged**